### PR TITLE
Add multibyte + zero-length pattern tests

### PR DESCRIPTION
**Pull request by vbkaisetsu**
This branch add test cases for patterns that includes both multibyte and empty strings.

### CODE REVIEW

File: src/charwise.rs [hunk 1]
```diff
@@ -1366,6 +1366,92 @@ mod tests {
         assert_eq!(0x100, pma.states[0x80].base().unwrap().get());
     }
 
+    #[test]
+    fn test_multibyte_overlapping_zerolength_pattern() {
+        let patterns = vec!["a", "æ", "あ", ""];
+        let pma = CharwiseDoubleArrayAhoCorasick::new(patterns).unwrap();
```
[the remark this answers is not in the document]
Comment (vbkaisetsu) [reply src/charwise.rs:1370]: output type does not affect the behavior of this automaton.

File: src/charwise.rs [hunk 2]
```diff
@@ -1366,6 +1366,92 @@ mod tests {
...
+                (16, 16, 3),
+            ],
+            results,
+        )
+    }
+
+    #[test]
+    fn test_multibyte_leftmost_zerolength_pattern() {
+        let patterns = vec!["a", "æ", "あ", ""];
+        let pma = CharwiseDoubleArrayAhoCorasickBuilder::new()
```
[the remark this answers is not in the document]
Comment (vbkaisetsu) [reply src/charwise.rs:1431]: output type does not affect the behavior of this automaton.

File: src/charwise.rs [hunk 3]
```diff
@@ -1366,6 +1366,92 @@ mod tests {
...
+        /*
+         *  い 0..3
+         *  あ 3..6
+         *  a  6..7
+         *  b  7..8
+         *  c  8..9
+         *  Æ  9..11
+         *  æ  11..13
+         *  う 13..16
+         */
```
Comment (kampersanda) [inline src/charwise.rs:1384]: why using `/* */` syntax?

Comment (vbkaisetsu) [reply src/charwise.rs:1384]: what do you mean?

Comment (vbkaisetsu) [reply src/charwise.rs:1384]: This is the same coding style as
https://github.com/daac-tools/daachorse/blob/main/src%2Fbytewise.rs#L1194-L1200

Comment (kampersanda) [reply src/charwise.rs:1384]: I see.

**Review by kampersanda — APPROVED** [review]
